### PR TITLE
Update corpora API

### PR DESCRIPTION
**Pull request by Bento007**
Updating the corpora API to match the hight level architecture.

### CODE REVIEW

File: data_portal_api.yml [hunk 1]
```diff
@@ -198,17 +332,28 @@ paths:
         A schema that contains a field dictating the visibility of the deployments that will either make the project
```
Comment (maniarathi) [inline data_portal_api.yml:335]: This doesn't seem correct 😆  Looks like in the API design doc, I just forgot to write a description for this API, but this is a description of something else.

File: data_portal_api.yml [hunk 2]
```diff
@@ -198,17 +332,28 @@ paths:
...
       parameters:
-        - $ref: '#/components/parameters/submission_uuid'
+        - $ref: '#/components/parameters/project_uuid'
       requestBody:
         content:
           application/json:
             schema:
               description: >-
-                a schema that has fields for all available project-level metadata.
+                a schema that contains a field dictating the visibility of the deployments that will either make the
```
Comment (maniarathi) [inline data_portal_api.yml:344]: I think you might have interpreted this too literally - the text was explaining what the parameters here should be which is literally a single enum dictating whether to publish the project privately or publicly.

File: data_portal_api.yml [hunk 3]
```diff
@@ -226,11 +377,6 @@ paths:
         This lists all projects and their UUIDs that currently exist in the data portal. If a parameter is specified as
         a filter, then only projects that meet the status criteria will be outputted.
       parameters:
-        - name: status
-          in: query
-          schema:
-            type: string
-            enum: [foo,bar]
         - $ref: '#/components/parameters/user_uuid'
         - name: fromDate
```
Comment (maniarathi) [inline data_portal_api.yml:381]: I think it's more conventional to use underscores? `from_date`? Ditto for toDate.

File: data_portal_api.yml [hunk 4]
```diff
@@ -57,16 +94,19 @@ paths:
...
       description: >-
-        This call will return all available metadata information about a project’s submission, including datasets that
-        are attached to the project.
+        Returns all available metadata information about a project submission, including URIs of datasets that are
+        attached to the project.
       parameters:
-        - $ref: '#/components/parameters/submission_uuid'
+        - $ref: '#/components/parameters/project_uuid'
       responses:
         '200':
```
[the remark this answers is not in the document]
Comment (Bento007) [reply data_portal_api.yml:115]: I think that is beyond the scope of this PR

File: data_portal_api.yml [hunk 5]
```diff
@@ -124,63 +203,118 @@ paths:
...
-          $ref: '#/components/responses/200'
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  project_uuid:
+                    $ref: '#/components/schemas/project_uuid'
+                  project:
```
[the remark this answers is not in the document]
Comment (Bento007) [reply data_portal_api.yml:308]: maybe title?

File: data_portal_api.yml (hunk 3 above)
Comment (Bento007) [reply data_portal_api.yml:373]: I actually thing for JSON the convention is camelcase over underscores.

File: data_portal_api.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (Bento007) [reply data_portal_api.yml:115]: I create a ticket to track the work

File: data_portal_api.yml (hunk 2 above)
Comment (Bento007) [reply data_portal_api.yml:338]: yeah, a bit too litteral on my part.

File: corpora-api.yml [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/projects:
```
Comment (calvinnhieu) [inline corpora-api.yml:367]: Noticed the API uses singular for `submission` and `dataset`, but plural for `projects`. Should these be consistent?

File: corpora-api.yml [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+          $ref: '#/components/responses/400'
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/submission/{project_uuid}/file:
```
Comment (calvinnhieu) [inline corpora-api.yml:148]: This endpoint accepts 3 body parameters: `filename`, `filetype`, `body`. Is the `body` intended to contain the encoded contents of the uploaded file? If so, I was wondering if we would consider an alternative approach to file upload, possible options proposed [here](https://stackoverflow.com/a/4083908).

In particular, I'm interested in the two-part upload option: 1) POST `/file_metadata` to create a file entry, and 2) POST `/file` via `multipart/form-data`, as a cleaner and possibly faster file upload pattern. Thoughts @Bento007 and @maniarathi ? Curious how the DCP Upload Service performed this operation as well.

File: corpora-api.yml [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+
+  /v1/submission/{project_uuid}/save:
+
+    post:
+      tags:
+        - submissions
+      summary: Save the project
+      description: >-
+        If needed (i.e. project level metadata was not inputted via a form and instead embedded in the matrix file),
+        extracts the project-level metadata from the datasets or uses the given project-level metadata imputed via the
```
Comment (calvinnhieu) [inline corpora-api.yml:276]: typo: imputed -> inputted

File: corpora-api.yml [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+  /v1/submission/{project_uuid}:
+
+    get:
```
Comment (calvinnhieu) [inline corpora-api.yml:105]: Just clarifying as I mentally walk through Data Portal user flows:

- this endpoint would be used to render a "private"/"in review" project
- GET `/v1/project/{project_uuid}` will be used to render a "public" project

Yes?

File: corpora-api.yml [hunk 10]
```diff
@@ -0,0 +1,530 @@
...
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+
+
+components:
+
+  schemas:
+    user_uuid:
```
Comment (calvinnhieu) [inline corpora-api.yml:472]: This was likely addressed during @maniarathi's high level design review, but we'll need a User API to implement and enforce authn/z in the future.

File: corpora-api.yml [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+      tags:
+        - dataset
+      summary: Request to download a dataset
+      description: >-
+        Request to download a file which on success will generate a pre-signed URL to download the dataset.
+      parameters:
+        - $ref: '#/components/parameters/path_dataset_uuid'
+      responses:
+        '200':
+          $ref: '#/components/responses/200'
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply corpora-api.yml:473]: Sorry - I don't seem to see the change here?

File: corpora-api.yml [hunk 12]
```diff
@@ -0,0 +1,530 @@
...
+      in: query
+      schema:
+        $ref: '#/components/schemas/user_uuid'
+
+  responses:
+    200:
+      description: OK.
+    202:
+      description: Accepted
+    201:
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply corpora-api.yml:527]: Thank you :D

File: corpora-api.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (Bento007) [reply corpora-api.yml]: now it's fixed

**Pull request merged**